src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -401,22 +401,27 @@ private boolean isNativeObjectType(JSType type, Compiler compiler) {
   }
 
   private boolean hasNestedTypes(JSType type) {
-    if (!isTrueConstructor(type)) return false;
+    if (!isClassLike(type)) return false;
     FunctionType ftype = (FunctionType) type;
     for (String name : ((FunctionType) type).getOwnPropertyNames()) {
       JSType propType = ftype.getPropertyType(name);
-      if (isTrueConstructor(propType) || propType.isEnumType() || propType.isInterface()
-          || isTypedef(propType)) return true;
+      if (isClassLike(propType) || propType.isEnumType() || propType.isInterface()
+          || isTypedef(propType)) {
+        return true;
+      }
     }
     return false;
   }
 
   /**
-   * Confusingly, the typedef type returns true on isConstructor checks, so we need to filter
-   * the NoType through this utility method.
+   * Returns true for types that are class like, i.e. that define a constructor for a nested type.
```

Comment:
Are interfaces and enums classlike (conceptually and according to the function check). Expand the comment - "constructor for a nested type" does not tell me enough to use this method, feels based on callsites not based on function body.